### ISSUE TITLE
aroundAll runs even when all specs are ignored

### ISSUE THREAD

**Issue by greghaskins** [greghaskins/spectrum#73]
This is a low-priority one, but should probably get addressed at some point. At the moment, there isn't a simple implementation fix, because `Suite` objects don't know which of their children are ignored or not. I think the underlying implementation will evolve to be based more around tags anyway; fixing this issue will be simpler after that.

For reference, this is the failing test:

```java
describe("a suite with only ignored specs", () -> {

  it("should not run aroundAll", () -> {
    ArrayList<String> steps = new ArrayList<>();
    SpectrumHelper.run(() -> {

      aroundAll(block -> {
        steps.add("aroundAll");
        block.run();
      });

      xit("foo", () -> {
      });

    });

    assertThat(steps, is(empty()));
  });

});
```

**Comment by ashleyfrieze** [comment]
The suite should be able to tell if its children are ignored. The question is, I guess, whether it needs to be testing for ignored of direct descendants, or the whole tree.

Just add `isIgnored` to `Child` and then 

```java
// suite
boolean areChildrenIgnored() {
    return children.stream().filter(child -> !child.isIgnored()).findFirst().isPresent();
}
``

**Comment by ashleyfrieze** [comment]
This will be fixed by #56 which would otherwise be doing big class-rule things with totally ignored suites.

**Comment by greghaskins** [comment]
@ashleyfrieze we should check if this bug still applies, or whether it was fixed by #56 .

**Comment by ashleyfrieze** [comment]
````java
@RunWith(Spectrum.class)
public class AroundAllSpec {{
	describe("A suite with an around all", () -> {
		aroundAll(block -> {
			System.out.println("Before running");
			block.run();
			System.out.println("after running");
		});
		
		xit("should not run", () -> {});
	});
}}
````

The above produces

````
Before running

Test ignored.
after running
````

**Comment by ashleyfrieze** [comment]
So the bug still applies

**Comment by ashleyfrieze** [comment]
I think this may have been fixed by #56 

````java
      describe("a suite with only ignored specs", () -> {

        it("should not run aroundAll", () -> {
          ArrayList<String> steps = new ArrayList<>();
          SpectrumHelper.run(() -> {

            aroundAll(block -> {
              steps.add("aroundAll");
              block.run();
            });

            xit("foo", () -> {
            });

          });

          assertThat(steps, is(empty()));
        });

      });
````

That passes - it's a test within a test. The `SpectrumHelper.run` wraps the real block with both ignored and `aroundAll`. The effect of running this is that the array list remains empty, so the hook didn't run.

I wonder if I stole this test from somewhere and made it pass while making the hooks work for JUnit mixins. I can't remember.

**Comment by greghaskins** [comment]
Resolved by #56.
